### PR TITLE
Fixed #31327 -- Deprecated providing_args argument for Signal.

### PR DESCRIPTION
**Pull request by jdufresne**
https://code.djangoproject.com/ticket/31327

### CODE REVIEW

File: docs/releases/3.1.txt [hunk 1]
```diff
@@ -543,6 +543,10 @@ Miscellaneous
   older versions of Django. Support for the old format remains until Django
   4.0.
 
+* The purely documentational ``providing_args`` argument for
+  :class:`~django.dispatch.Signal` is deprecated. If you rely on this
+  argument as documentation, you can move the text to a code comment.
+
```
Comment (carltongibson) [inline docs/releases/3.1.txt]:
```suggestion
  argument as documentation, you can move the text to a code comment, or 
  docstring.
  
```

File: django/dispatch/dispatcher.py [hunk 2]
```diff
@@ -28,14 +30,16 @@ class Signal:
...
         self.receivers = []
-        if providing_args is None:
-            providing_args = []
-        self.providing_args = set(providing_args)
+        if providing_args is not None:
+            warnings.warn(
+                'The providing_args argument is deprecated. As it is purely '
+                'documentational, it has no replacement. If you rely on this '
+                'argument as documentation, you can move the text to a code '
+                'comment.',
```
[the remark this answers is not in the document]
Comment (jdufresne) [reply django/dispatch/dispatcher.py]: The edits, LGTM. Although I don't think we need the added comma as there are only two items. 

Would you like me to push them or do you prefer to do so upon merge?

[the remark this answers is not in the document]
Comment (carltongibson) [reply django/dispatch/dispatcher.py]: Happy for you to do it. 😃

**Comment by jdufresne** [comment]
Done. Thanks for the review @carltongibson

**Comment by carltongibson** [comment]
Nice work @jdufresne. 🧹

**Pull request merged**